eth2/_utils/bls/validation.py

```diff
@@ -3,21 +3,46 @@
 )
 from eth_typing import (
     BLSSignature,
+    BLSPubkey,
 )
-from eth_utils import (
-    ValidationError,
+from eth2.beacon.constants import (
+    EMPTY_PUBKEY,
+    EMPTY_SIGNATURE,
 )
+from eth2.beacon.exceptions import (
+    SignatureError,
+    PublicKeyError,
+)
+from typing import Sequence
 
 
 def validate_private_key(privkey: int) -> None:
     if privkey <= 0 or privkey >= curve_order:
-        raise ValidationError(
+        raise ValueError(
             f"Invalid private key: Expect integer between 1 and {curve_order - 1}, got {privkey}"
         )
 
 
+def validate_public_key(pubkey: BLSPubkey, allow_empty: bool =False) -> None:
+    if len(pubkey) != 48:
+        raise PublicKeyError(
+            f"Invalid public key length, expect 48 got {len(pubkey)}. pubkey: {pubkey}"
+        )
+    if not allow_empty and pubkey == EMPTY_PUBKEY:
+        raise PublicKeyError(f"Empty public key is invalid  pubkey={pubkey}")
+
+
+def validate_many_public_keys(pubkeys: Sequence[BLSPubkey]) -> None:
```

Comment:
Maybe remove `many`:
```suggestion
def validate_public_keys(pubkeys: Sequence[BLSPubkey]) -> None:
```

Reply:
`validate_public_keys` would look very similar to `validate_public_key`, not sure if we want that. That's why making it `validate_many_public_keys`.